nextcord/polls.py

```diff
@@ -0,0 +1,448 @@
+    # https://github.com/nextcord/nextcord/blob/master/nextcord/message.py#L68
+    EmojiInputType = Union[Emoji, PartialEmoji, str]
+
+__all__ = (
+    "PollResults",
+    "PollMedia",
+    "Poll",
+    "PollAnswerCount",
+    "PollCreateRequest",
+    "PollAnswer",
+)
+
+
+def resolve_emoji(emoji: Union[Emoji, PartialEmoji]) -> PartialEmojiPayload:
+    if isinstance(emoji, Emoji):
+        return cast(PartialEmojiPayload, emoji._to_partial().to_dict())
+
+    if isinstance(emoji, PartialEmoji):
+        return cast(PartialEmojiPayload, emoji.to_dict())
+
+    raise InvalidArgument(f"Cannot parse `{emoji}` into an emoji to send.")
+
+
+class PollMedia:
+    """A common object that backs both the question and answer.
+
+    .. versionadded:: 3.0
+
+    Attributes
+    ----------
+    text: Optional[:class:`str`]
+        The text of the media.
+    emoji: Optional[Union[:class:`PartialEmoji`, :class:`Emoji`]]
+        The emoji of the media.
+
+    """
+
+    __slots__ = (
+        "text",
+        "emoji",
+    )
+
+    def __init__(self, text: str, emoji: Optional[EmojiInputType] = None) -> None:
+        self.text: str = text
+        self.emoji: Optional[PartialEmoji | Emoji] = emoji  # type: ignore[reportAttributeAccessIssue]
+        if isinstance(emoji, str):
+            self.emoji: Optional[PartialEmoji | Emoji] = PartialEmoji.from_str(emoji)
+
+    def __repr__(self) -> str:
+        return f"<PollMedia text={self.text} emoji={self.emoji}>"
+
+    def to_dict(self) -> PollMediaPayload:
+        payload: PollMediaPayload = {"text": self.text}
+        if self.emoji:
+            payload["emoji"] = resolve_emoji(self.emoji)
+
+        return payload
+
+    @classmethod
+    def from_dict(cls, data: PollMediaPayload) -> Self:
+        emoji = PartialEmoji.from_dict(data["emoji"]) if "emoji" in data else None
+
+        return cls(text=data["text"], emoji=emoji)
+
+
+class PollAnswer:
+    """A choice to answer in a poll.
+
+    .. versionadded:: 3.0
+
+    Attributes
+    ----------
+    answer_id: Optional[:class:`int`]
+        The answer ID of the answer.
+    poll_media: Optional[:class:`PollMedia`]
+        The poll media of the answer.
+    """
+
+    __slots__ = (
+        "answer_id",
+        "poll_media",
+    )
+
+    def __init__(self, *, answer_id: Optional[int] = None, poll_media: PollMedia) -> None:
+        self.answer_id: Optional[int] = answer_id
+        self.poll_media: PollMedia = poll_media
+
+    def __repr__(self) -> str:
+        return f"<PollAnswer answer_id={self.answer_id!r} poll_media={self.poll_media!r}>"
+
+    def to_dict(self) -> PollAnswerPayload:
+        payload: PollAnswerPayload = {"poll_media": self.poll_media.to_dict()}
+        if self.answer_id:
+            payload["answer_id"] = self.answer_id
+
+        return payload
+
+    @classmethod
+    def from_dict(cls, data: PollAnswerPayload) -> Self:
+        # data['answer_id'] is always sent as part of Discord's HTTP/Gateway
```

Comment:
I think an alternative to get around this problem is to make another typed dict specifically for poll answer payloads from Discord API requests.

Reply:
I don't really think there's an issue with that, since `to_dict` and `from_dict` are used in different situations.